silabs_flasher/rootfs/etc/s6-overlay/scripts/universal-silabs-flasher-up

```diff
@@ -51,7 +51,7 @@ if bashio::config.has_value 'firmware_url'; then
 else
     # Assume to run on Yellow if UART4 is mapped to ttyAMA1
     if [ -d /sys/devices/platform/soc/fe201800.serial/tty/ttyAMA1 ] && [ "${device}" == "/dev/ttyAMA1" ]; then
-        firmware="NabuCasa_Yellow_EZSP_v7.2.2.0_PA32_ncp-uart-hw_115200.gbl"
+        firmware="NabuCasa_Yellow_EZSP_v7.3.1.0_PA32_ncp-uart-hw_115200.gbl"
     else
         # Check device manufacturer/product information
         usb_device_path=$(realpath /sys/class/tty/$(readlink /sys/class/tty/$(basename ${device}))/../../../../)
```

Comment:
I ended up using the following here, so it correctly detects the USB device on all drivers (for e.g. CDC), although probably doesnt make much difference for the use case here, it does show up in the logs (here and multiprotocol) "checking device identifying "some USB Hub identifier".
```
usb_device_path=$(realpath /sys/class/tty/$(readlink /sys/class/tty/$(basename ${device}) | sed 's:/[^/]*tty[^/]*::g' )/../)
```

Reply:
Hm, using sed to cut the dev path seems a bit hacky too, but I guess its the best we can do :man_shrugging:

Reply:
yeh it feels like there should be a cleaner way to just get the usb device path, but I couldnt find one...